eduNEXT/eox-tenant
# feat: adds FilterTenantAwareLinksFromStudio in filters pipeline

## Description

This PR adds a custom filter pipeline to get LMS_ROOT_URL value for org from tenant site.

The idea is including some changes in _openedx-filters_ and _edx-platform_ repositories to anyone implement the filter function, and one clear example to use is this custom pipeline on eox-tenant plugin.

**Dependencies**:
- openedx/edx-platfom: https://github.com/openedx/edx-platform/pull/35142
- openedx/openedx-filters: https://github.com/openedx/openedx-filters/pull/200
- 
## Testing instructions

1. Install the plugin with the changes in [openedx/edx-platform](https://github.com/openedx/edx-platform/pull/35142) and [openedx/openedx-filters](https://github.com/openedx/openedx-filters/pull/200) branches. Use the [Tutor instructions](https://docs.tutor.overhang.io/configuration.html#installing-extra-requirements-from-private-repositories) to install the eox-tenant django plugin with the changes in this branch.

2. Create a Tenant and import a certificate course (You need to create a tenant for Studio too). Go on to create sites: http://local.edly.io:8000/admin/eox_tenant/tenantconfig/

3. Add the following configurations to your configuration site and Studio tenant.
```json
"OPEN_EDX_FILTERS_CONFIG": {
                "org.openedx.learning.course_about.page.url.requested.v1": {
                    "fail_silently": false,
                    "pipeline": [
                        "eox_tenant.filters.pipeline.OrgAwareCourseAboutPageURL"
                    ]
                },
                "org.openedx.course_authoring.lms.page.url.requested.v1": {
                    "fail_silently": false,
                    "pipeline": [
                        "eox_tenant.filters.pipeline.OrgAwareLMSURLStudio"
                    ]
                }
            }
```
or too you can add settings in `/env/apps/openedx/settings/cms/development.py` 

4. Go to Settings-> Schedule & Details, click on the “Course Summary Page” link, which should direct you to the URL of the specific tenant
![Captura de pantalla de 2024-10-01 23-39-30](https://github.com/user-attachments/assets/3e529d68-7107-4aa0-bb23-c079b253bbca)

5. Verify that the link of the "Invite your students" button in the body refers to the URL of the specific tenant

6. Also check for a link to a course summary below the “course overview” text box.
![Captura de pantalla de 2024-10-01 23-40-55](https://github.com/user-attachments/assets/9ad8f396-be1f-441f-95a9-651b83f7604c)

7. Go to Content -> Files & uploads, verify the absolute links of any asset (In the Web button), which should lead to the URL of the specific tenant

### Example Tenant Config

```json
{
    "EDNX_USE_SIGNAL": true,
    "LMS_BASE": "tenant-a.local.edly.io:8000",
    "LMS_ROOT_URL": "tenant-a.local.edly.io:8000",
    "PLATFORM_NAME": "TEST-THEMING",
    "SITE_NAME": "tenant-a.local.edly.io",
    "THEME_OPTIONS": {
        "scripts": {
            "/login": [
                {
                    "content": "alert('This is a test for the inline script');",
                    "type": "inline"
                }
            ]
        },
        "theme": {
            "grandparent": "test-3",
            "name": "test-1",
            "parent": "test-2"
        }
    },
    "course_org_filter": [
        "DEMO"
    ]
}
```

## Additional information

- Test cases file: https://docs.google.com/document/d/1lWlC1_xqO308oHNH0F4fOdyV3uBlVte0wQC6UOa_3SE/edit?usp=sharing

## Checklist for Merge
